File: 1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:

        def traversal(root):
            if root==None:
                return
            if root.val==target.val:
                return root
            left=traversal(root.left)
            right=traversal(root.right)
            if left!=None:
                return left
            else:
                return right
        #----------------------------
        return traversal(cloned)

File: 1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    public TreeNode dfs(TreeNode root,TreeNode target){
        if(root==null){
            return null;
        }
        if(root.val==target.val){
            return root;
        }
        TreeNode left=dfs(root.left,target);
        TreeNode right=dfs(root.right,target);
        if(left!=null){
            return left;
        }
        return right;
    }
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        return dfs(cloned,target);
    }
}